src/transformers/models/swin/modeling_swin.py

```diff
@@ -0,0 +1,915 @@
+        self.key = nn.Linear(self.all_head_size, self.all_head_size, bias = config.qkv_bias)
+        self.value = nn.Linear(self.all_head_size, self.all_head_size, bias = config.qkv_bias)
+
+        self.dropout = nn.Dropout(config.attn_drop_rate)
+
+    def transpose_for_scores(self, x):
+        new_x_shape = x.size()[:-1] + (self.num_attention_heads, self.attention_head_size)
+        x = x.view(*new_x_shape)
+        return x.permute(0, 2, 1, 3)
+
+    def forward(
+        self,
+        hidden_states,
+        attention_mask=None,
+        head_mask=None,
+        output_attentions=False,
+    ):
+        batch_size, dim, num_channels = hidden_states.shape
+        mixed_query_layer = self.query(hidden_states)
+
+        key_layer = self.transpose_for_scores(self.key(hidden_states))
+        value_layer = self.transpose_for_scores(self.value(hidden_states))
+        query_layer = self.transpose_for_scores(mixed_query_layer)
+
+        # Take the dot product between "query" and "key" to get the raw attention scores.
+        attention_scores = torch.matmul(query_layer, key_layer.transpose(-1, -2))
+
+        attention_scores = attention_scores / math.sqrt(self.attention_head_size)
+
+        relative_position_bias = self.relative_position_bias_table[self.relative_position_index.view(-1)].view(
+            self.window_size[0] * self.window_size[1], self.window_size[0] * self.window_size[1], -1) 
+
+        relative_position_bias = relative_position_bias.permute(2, 0, 1).contiguous()  
+        attention_scores = attention_scores + relative_position_bias.unsqueeze(0)
+
+        if attention_mask is not None:
+            # Apply the attention mask is (precomputed for all layers in SwinModel forward() function)
+            nW = attention_mask.shape[0]
+            attention_scores = attention_scores.view(batch_size // nW, nW, self.num_attention_heads, dim, dim) + attention_mask.unsqueeze(1).unsqueeze(0)
+            attention_scores = attention_scores.view(-1, self.num_attention_heads, dim, dim)
+
+        # Normalize the attention scores to probabilities.
+        attention_probs = nn.functional.softmax(attention_scores, dim=-1)
+
+        # This is actually dropping out entire tokens to attend to, which might
+        # seem a bit unusual, but is taken from the original Transformer paper.
+        attention_probs = self.dropout(attention_probs)
+
+        # Mask heads if we want to
+        if head_mask is not None:
+            attention_probs = attention_probs * head_mask
+
+        context_layer = torch.matmul(attention_probs, value_layer)
+        context_layer = context_layer.permute(0, 2, 1, 3).contiguous()
+        new_context_layer_shape = context_layer.size()[:-2] + (self.all_head_size,)
+        context_layer = context_layer.view(*new_context_layer_shape)
+
+        outputs = (context_layer, attention_probs) if output_attentions else (context_layer,)
+
+        return outputs 
+
+class SwinSelfOutput(nn.Module):
```

Comment:
```suggestion
# Copied from transformers.models.vit.modeling_vit.ViTSelfOutput with ViT->Swin
class SwinSelfOutput(nn.Module):
```

Reply:
Why close this? the comment is missing.

Reply:
I closed this, and some other reviews regarding the copies, because `SwinSelfOutput` is not an exact copy of `VitSelfOutput`. Unlike ViT, most classes in Swin need a `dim` passed to them while initialization.